idempotence4j-core/src/main/java/com/transferwise/idempotence4j/core/retention/RetentionPolicy.java

```diff
@@ -4,27 +4,45 @@
 import com.cronutils.model.definition.CronDefinitionBuilder;
 import com.cronutils.parser.CronParser;
 import com.cronutils.utils.Preconditions;
+import com.github.kagkarlsson.scheduler.task.schedule.Schedule;
+import com.github.kagkarlsson.scheduler.task.schedule.Schedules;
 import lombok.Getter;
 import lombok.NonNull;
 import lombok.extern.slf4j.Slf4j;
 
+import java.time.Duration;
 import java.time.Period;
+import java.time.temporal.TemporalAmount;
 
 @Getter
 public class RetentionPolicy {
-    private final Period period;
+    private final TemporalAmount temporalAmount;
     private final PurgeJobConfiguration purgeJobConfiguration;
 
     public RetentionPolicy(
-        @NonNull Period period,
+        String period,
+        String duration,
         @NonNull PurgeJobConfiguration purgeJobConfiguration
     ) {
-        Preconditions.checkArgument(!(period.isNegative() || period.isZero()), "Retention period has to be positive");
+        if (period != null && duration == null) {
+            Period p = Period.parse(period);
+            this.temporalAmount = p;
+            Preconditions.checkArgument(!(p.isNegative() || p.isZero()), "Retention period has to be positive or null");
+        } else if (period == null && duration != null) {
+            Duration d = Duration.parse(duration);
+            this.temporalAmount = d;
+            Preconditions.checkArgument(!(d.isNegative() || d.isZero()), "Retention duration has to be positive or null");
+        } else {
```

Comment:
can we have 2 constructors instead?

Reply:
I also do not get the need for wrong naming of local variables, `p` and `d` doesn't carry any context.

Reply:
- 2 constructors: it just pushes the conditional logic to the caller.
- `p, d`: the scope of the local variables is 3 lines but we can make it 2, i will check

Reply:
> * 2 constructors: it just pushes the conditional logic to the caller.

it's also on caller (the caller should define one or another parameter only), but two constructors will just make it more defined and explicit

Reply:
But this is a Flag argument code smell, those 2 fields decide the direction,  splitting these 2 into different function with proper naming might be a good first improvement.
1. checkPreConditionsIfOnlyPeriodAvailable(...)
2. checkPreConditionsIfOnlyDurationAvailable(...)

Also,
I feel that the null field checks should have a separate function, that only decides what type to check.  Maybe create an Enum 
Say:
`public enum WhateverName {
      PERIOD_ONLY, DURATION_ONLY;
}`


Then:
Move the `if (period != null && duration == null)` && `if (period == null && duration != null)` into:

private WhateverName checkDirection(String period, String duration) {
if (period != null && duration == null) {
         return PERIOD_ONLY;
     } else if (period == null && duration != null) {
         return DURATION_ONLY;
    }
}

Reply:
it feels like overengineering to me.

Reply:
Hey, i am not blocking tbh, i dont think it is over engineering, it is just clarifying and making the code look a lot cleaner, i am seeing this code for the first time and had to look multiple times to understand.